pkgs/build-support/build-fhsenv-bubblewrap/default.nix

```diff
@@ -192,6 +192,8 @@ let
       ${lib.optionalString unshareCgroup "--unshare-cgroup"}
       ${lib.optionalString dieWithParent "--die-with-parent"}
       --ro-bind /nix /nix
+      --ro-bind /etc /.host-etc
+      --tmpfs /tmp
```

Comment:
This should be configurable (i.e. `privateTmp`). Probably even disabled by default, given that it's just steam with its asinine electron cookies clearing that really needs this.

Remember: Lots of things other than Steam use buildFHSEnv.

Reply:
I did it for `buildFHSEnv` generally in part because in an earlier conversion with @jonringer about implementing this, he suggested that it should probably be that way for everyone.

I'm not opposed to doing it just for steam, but I do see the sensibility of separating `/tmp` for all FHS envs. I think this probably requires a bit more discussion.

Reply:
I made it configurable, but on by default for now. More discussion may be needed to figure out what the default should be.

Reply:
The biggest problem I see with private tmp by default is that this is mostly used for GUI user applications, many of which include a file picker. If the user navigates to `/tmp` or other "user data" directories, they expect to find the "host" system's data in it.

Remember that, eventhough it uses sandboxing to achieve its purpose, buildFHSEnv is a compatibility tool, not a sandboxing one; we want as little difference to the "host" system as possible.

In the case of Steam, we have a case of it doing stupid things it shouldn't to work around. Unless such a case exists, we should follow the principle of least surprise and not sandbox `/tmp`.  
Though even with Steam I'm hesitant. You already had to make an exception for dumps. As soon as we work around undesired behaviour using such a hammer approach, it's now our responsibility that all desired features still work.  
It's a bit contrived but imagine a user creating a temporary Steam library inside `/tmp` in order to muck around with some game. If they needed to modify game files, it'd be hella confusing to not find your game library in `/tmp` and I don't even see an easy way to access it otherwise.

Reply:
A counter-argument, for me, is that proper use of `buildFHSEnv` already pretty much requires properly set up xdg portals, so that opening things with `xdg-open` and the like doesn't run the programs inside the FHS env, causing various forms of havoc.

Even if the purpose isn't sandboxing, it necessarily comes with many of the same context management issues that sandboxing environments do. This hybrid sandbox-but-also-not environment we're using now has a lot of rough edges like that, and I feel like leaning in the direction of more sandboxing will be moving with the grain of what the relevant applications are supporting anyway (i.e. flatpak).

Reply:
We could mount the host's `/tmp` at another location like `/host-tmp`. That would at least be reasonably self-explanatory in the case where the user is manually looking for a file, and for someone doing something more involved and scripted, they at least have an option to access it.

Reply:
> proper use of `buildFHSEnv` already pretty much requires properly set up xdg portals, so that opening things with `xdg-open` and the like doesn't run the programs inside the FHS env, causing various forms of havoc.

This is not by choice. If it could be done differently, buildFHSEnv would do so.

> This hybrid sandbox-but-also-not environment we're using now has a lot of rough edges like that, and I feel like leaning in the direction of more sandboxing will be moving with the grain of what the relevant applications are supporting anyway (i.e. flatpak).

If anything, these aid to smooth these rough edges over. Running unpatched binaries will never be entirely seamless but we should aim to be as seamless as possible given the constraints.

Explicitly adding seams does not work towards that goal. If we wanted flatpak, we might aswell use flatpak.

Reply:
Ok, yeah. After some thought, I agree.

The goal of `buildFHSEnv` is compatibility, and isolating `/tmp` is not a sensible default from the perspective of that goal.

Reply:
I agree with @atemu, the majority of these use cases are for usability.